### ISSUE TITLE
Add base bot command object

### ISSUE THREAD

**Issue by hsoj** [hsoj/dubilier#13]
### Context

In order to provide an interface that can easily be discovered, as well as implemented, a base object that provides the required functionality for a bot command to be created is needed.  The `dubilier.bot.discord.Daemon` object should provide a method to include a command into the bot's awareness.

### Out of scope

* Not expected to actually be callable, nor visible on the list of available commands the user can execute.

### Acceptance criteria

* Base object can be initialized
* Daemon object provides a method to add the command

### Technical notes

Previously the `Cog` object was inherited for this purpose.  Should determine if this is still the best approach, and if not, inherit from a more reasonable object if it makes sense to follow this pattern.

**Comment by hsoj** [comment]
In order to use the object pattern, it appears that the `Cog` object provides the capability.

**Comment by hsoj** [comment]
Blocked by refactor of Daemon object: #15

**Comment by hsoj** [comment]
The base command object was used to demonstrate the acceptance criteria of #15
